src/PowerShellEditorServices/Extensions/Api/EditorUIService.cs

```diff
@@ -0,0 +1,171 @@
+        /// </summary>
+        /// <param name="message">The message to display for the prompt.</param>
+        /// <param name="choices">The choices to give the user.</param>
+        /// <returns>A list of the labels of selected choices, or null if the prompt was canceled.</returns>
+        Task<IReadOnlyList<string>> PromptMultipleSelection(string message, IReadOnlyList<PromptChoiceDetails> choices);
+
+        /// <summary>
+        /// Prompt a set of selections from a list of choices.
+        /// </summary>
+        /// <param name="message">The message to display for the prompt.</param>
+        /// <param name="choices">The choices to give the user.</param>
+        /// <param name="defaultChoiceIndexes">A list of the indexes of choices to mark as default.</param>
+        /// <returns>A list of the labels of selected choices, or null if the prompt was canceled.</returns>
+        Task<IReadOnlyList<string>> PromptMultipleSelection(string message, IReadOnlyList<PromptChoiceDetails> choices, IReadOnlyList<int> defaultChoiceIndexes);
+    }
+
+    internal class EditorUIService : IEditorUIService
+    {
+        private static string[] s_choiceResponseLabelSeparators = new[] { ", " };
+
+        private readonly ILanguageServer _languageServer;
+
+        public EditorUIService(ILanguageServer languageServer)
+        {
+            _languageServer = languageServer;
+        }
+
+        public Task<IReadOnlyList<string>> PromptMultipleSelection(string message, IReadOnlyList<PromptChoiceDetails> choices) =>
+            PromptMultipleSelection(message, choices, defaultChoiceIndexes: null);
+
+        public async Task<IReadOnlyList<string>> PromptMultipleSelection(string message, IReadOnlyList<PromptChoiceDetails> choices, IReadOnlyList<int> defaultChoiceIndexes)
+        {
+            ChoiceDetails[] choiceDetails = GetChoiceDetails(choices);
+
+            ShowChoicePromptResponse response = await _languageServer.SendRequest<ShowChoicePromptRequest, ShowChoicePromptResponse>(
+                "powerShell/showChoicePrompt",
+                new ShowChoicePromptRequest
+                {
+                    IsMultiChoice = true,
+                    Caption = string.Empty,
+                    Message = message,
+                    Choices = choiceDetails,
+                    DefaultChoices = defaultChoiceIndexes?.ToArray(),
+                });
+
+            if (response.PromptCancelled)
+            {
+                return null;
+            }
+
+            return response.ResponseText.Split(s_choiceResponseLabelSeparators, StringSplitOptions.None);
+        }
+
+        public Task<string> PromptSelection(string message, IReadOnlyList<PromptChoiceDetails> choices) =>
+            PromptSelection(message, choices, defaultChoiceIndex: -1);
+
+        public async Task<string> PromptSelection(string message, IReadOnlyList<PromptChoiceDetails> choices, int defaultChoiceIndex)
```

Comment:
I deliberately took it out since it makes no sense. We just take two strings and concatenate them. So rather than bake that into the api, I think it’s best for us to fix the client

Reply:
Yeah probably a good call.  FWIW the origin of it was to match `$Host.UI.PromptForChoice`, since before the integrated console was a thing that was wired up to quick pick.